### PR TITLE
Make PreConfiguredTokenFilter harder to misuse

### PR DESCRIPTION
**Pull request by nik9000**
There are now three public static method to build instances of
PreConfiguredTokenFilter and the ctor is private. I chose static
methods instead of constructors because those allow us to change
out the implementation returned if we so desire.

Also moves over two more pre-configured token filters. There still
a bunch to go!

Relates to #23658

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilter.java [hunk 1]
```diff
@@ -36,31 +37,46 @@
  * Provides pre-configured, shared {@link TokenFilter}s.
  */
 public final class PreConfiguredTokenFilter implements AnalysisModule.AnalysisProvider<TokenFilterFactory> {
+    /**
+     * Set up a pre-configured token filter that may no vary at all.
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilter.java]: typo: no -> not

Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilter.java]: nit: "Set up" -> "Create"

**Review by rjernst — APPROVED** [review]
LGTM

**Review by jpountz — APPROVED** [review]

**Comment by nik9000** [comment]
From a categorization perspective this issue is about the plugin API more than analysis so I dropped the `:Analysis` label.

**Pull request merged**